File: apps/admin-v2/src/application/use-cases/service-logs/get-service-logs.use-case.ts
import type { ServiceLogRepository } from '@/domain/repositories/service-log.repository';
import type { ServiceLogFilters } from '@/domain/entities/service-log';

export class GetServiceLogsUseCase {
  constructor(private repo: ServiceLogRepository) {}

  execute(filters: ServiceLogFilters) {
    return this.repo.getAll(filters);
  }
}

File: apps/admin_mobile/lib/application/use_cases/service_logs/get_service_logs_use_case.dart
import '../../../domain/entities/service_log.dart';
import '../../../domain/repositories/service_log_repository.dart';
import '../../../shared/types/paginated_result.dart';

class GetServiceLogsUseCase {
  final ServiceLogRepository _repo;
  GetServiceLogsUseCase(this._repo);

  Future<PaginatedResult<ServiceLog>> call({String? date, int? page}) =>
      _repo.getAll(date: date, page: page);
}
